spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -139,105 +143,121 @@ public void testDeleteFileThenMetadataDelete() throws Exception {
   }
 
   @Test
   public void testDeleteFromEmptyTable() {
+    Assume.assumeFalse("Custom branch does not exist for empty table", "test".equals(branch));
     createAndInitUnpartitionedTable();
 
-    sql("DELETE FROM %s WHERE id IN (1)", tableName);
-    sql("DELETE FROM %s WHERE dep = 'hr'", tableName);
+    sql("DELETE FROM %s WHERE id IN (1)", commitTarget());
+    sql("DELETE FROM %s WHERE dep = 'hr'", commitTarget());
 
     Table table = validationCatalog.loadTable(tableIdent);
     Assert.assertEquals("Should have 2 snapshots", 2, Iterables.size(table.snapshots()));
 
     assertEquals(
         "Should have expected rows",
         ImmutableList.of(),
-        sql("SELECT * FROM %s ORDER BY id", tableName));
+        sql("SELECT * FROM %s ORDER BY id", selectTarget()));
+  }
+
+  @Test
+  public void testDeleteFromNonExistingCustomBranch() {
+    Assume.assumeTrue("Test only applicable to custom branch", "test".equals(branch));
+    createAndInitUnpartitionedTable();
+
+    Assertions.assertThatThrownBy(() -> sql("DELETE FROM %s WHERE id IN (1)", commitTarget()))
+        .isInstanceOf(ValidationException.class)
+        .hasMessage("Cannot operate against non-existing branch: test");
   }
 
   @Test
   public void testExplain() {
     createAndInitUnpartitionedTable();
 
     sql("INSERT INTO TABLE %s VALUES (1, 'hr'), (2, 'hardware'), (null, 'hr')", tableName);
+    createBranch();
 
-    sql("EXPLAIN DELETE FROM %s WHERE id <=> 1", tableName);
+    sql("EXPLAIN DELETE FROM %s WHERE id <=> 1", commitTarget());
 
-    sql("EXPLAIN DELETE FROM %s WHERE true", tableName);
+    sql("EXPLAIN DELETE FROM %s WHERE true", commitTarget());
 
     Table table = validationCatalog.loadTable(tableIdent);
     Assert.assertEquals("Should have 1 snapshot", 1, Iterables.size(table.snapshots()));
 
     assertEquals(
         "Should have expected rows",
         ImmutableList.of(row(1, "hr"), row(2, "hardware"), row(null, "hr")),
-        sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", tableName));
+        sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", commitTarget()));
   }
 
   @Test
   public void testDeleteWithAlias() {
     createAndInitUnpartitionedTable();
 
     sql("INSERT INTO TABLE %s VALUES (1, 'hr'), (2, 'hardware'), (null, 'hr')", tableName);
+    createBranch();
 
-    sql("DELETE FROM %s AS t WHERE t.id IS NULL", tableName);
+    sql("DELETE FROM %s AS t WHERE t.id IS NULL", commitTarget());
 
     assertEquals(
         "Should have expected rows",
         ImmutableList.of(row(1, "hr"), row(2, "hardware")),
-        sql("SELECT * FROM %s ORDER BY id", tableName));
+        sql("SELECT * FROM %s ORDER BY id", selectTarget()));
   }
 
   @Test
   public void testDeleteWithDynamicFileFiltering() throws NoSuchTableException {
     createAndInitPartitionedTable();
 
-    append(new Employee(1, "hr"), new Employee(3, "hr"));
+    append(tableName, new Employee(1, "hr"), new Employee(3, "hr"));
+    createBranch();
```

Comment:
Is it worth supporting `createBranch` when there's no table data? We could create an empty snapshot and branch automatically.

Reply:
Updated the method name. Regarding creating an empty snapshot, I remember we had this discussion before and ended up not doing that. I cannot recall the reason anymore, maybe we can hash out this in a different thread. Let me open an issue.